client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js

```diff
@@ -0,0 +1,249 @@
+import FormTextInput from 'components/forms/form-text-input';
+import Notice from 'components/notice';
+import QueryMailChimpSyncStatus from 'woocommerce/state/sites/settings/email/querySyncStatus';
+import {
+	syncStatus,
+	mailchimpSettings,
+	isRequestingSettings
+	} from 'woocommerce/state/sites/settings/email/selectors';
+import { submitMailchimpNewsletterSettings, requestResync } from 'woocommerce/state/sites/settings/email/actions.js';
+import { isSubmittingNewsletterSetting, newsletterSettingsSubmitError } from 'woocommerce/state/sites/settings/email/selectors';
+import { errorNotice, successNotice } from 'state/notices/actions';
+
+const SyncTab = localize( ( { siteId, translate, syncState, resync } ) => {
+	const { account_name, store_syncing, product_count, mailchimp_total_products,
+		mailchimp_total_orders, order_count } = syncState;
+	const hasProductInfo = ( product_count !== undefined ) && ( mailchimp_total_products !== undefined );
+	const products = hasProductInfo ? ( product_count + '/' + mailchimp_total_products ) : '';
+	const hasOrdersInfo = ( order_count !== undefined ) && ( mailchimp_total_orders !== undefined );
+	const orders = hasOrdersInfo ? ( order_count + '/' + mailchimp_total_orders ) : '';
+
+	const synced = () => (
+		<Notice
+			status="is-success"
+			isCompact
+			showDismiss={ false }
+			text={ translate( '%(mailingListname)s list synced.', {
+				args: { mailingListname: syncState.mailchimp_list_name } } ) }
+		/>
+	);
+
+	const syncing = () => (
+		<Notice
+			status="is-warning"
+			isCompact
+			showDismiss={ false }
+			text={ translate( '%(mailingListname)s list is being synced.', {
+				args: { mailingListname: syncState.mailchimp_list_name } } ) }
+		/>
+	);
+
+	const onResyncClick = () => {
+		! store_syncing && resync( siteId );
+	};
+
+	return (
+		<div>
+			<div>
+				<span className="mailchimp__account-info">{ translate( 'MailChimp account:' ) }</span>
+				<span>{ account_name }</span>
+			</div>
+			<span>{ store_syncing ? syncing() : synced() }</span>
+			<a onClick={ onResyncClick }>{ translate( 'Resync' ) }</a>
+			<div>
+				<span className="mailchimp__account-info" >{ translate( 'Products:' ) }</span>
+				<span>{ products }</span>
+				<span className="mailchimp__account-info" >{ ' ' + translate( 'Orders:' ) }</span>
+				<span>{ orders || '' }</span>
+			</div>
+		</div>
+	);
+} );
+
+const Settings = localize( ( { translate, settings, oldCheckbox, isSubbmittingSettings, onChange, onSave } ) => {
+	const onRadioChange = ( e ) => {
+		onChange( { mailchimp_checkbox_defaults: e.target.value } );
+	};
+
+	const onNewslatterLabelChange = ( e ) => {
+		onChange( { newsletter_label: e.target.value } );
+	};
+
+	const onToggle = ( e ) => {
+		const visibleOption = oldCheckbox !== 'hide' ? oldCheckbox : 'check';
+		onChange( { mailchimp_checkbox_defaults: e ? visibleOption : 'hide' } );
+	};
+
+	const checkbox = settings.mailchimp_checkbox_defaults;
```

Comment:
This variable name is opaque to me. Could it be something more like subscriptionPromptState (check, uncheck, none)

And then maybe instead of the similarly opaque `toggle` you could have something like showSubscriptionPrompt